Client/Frontend/Widgets/BottomSheetViewController.swift

```diff
@@ -188,27 +161,57 @@ class BottomSheetViewController: UIViewController {
                 $0.bottom.equalToSuperview()
             }
         }
+        
+        super.updateViewConstraints()
+    }
+    
+    // MARK: - Constraints setup
+    
+    private func makeConstraints() {
+        backgroundOverlayView.snp.makeConstraints {
+            let bottomInset = parent?.view.safeAreaInsets.bottom ?? 0
+            $0.leading.trailing.top.equalToSuperview()
+            $0.bottom.equalToSuperview().offset(bottomInset)
+        }
+        
+        handleView.snp.remakeConstraints {
+            $0.width.equalTo(UX.handleWidth)
+            $0.height.equalTo(UX.handleHeight)
+
+            $0.centerX.equalTo(contentView)
+            $0.top.equalTo(contentView).offset((UX.handleMargin - UX.handleHeight) / 2)
+        }
+        
+        closeButton.snp.makeConstraints {
+            $0.top.equalToSuperview().inset(10)
+            $0.right.equalToSuperview().inset(7)
+            $0.size.equalTo(26)
+        }
     }
     
     // MARK: - Animations
     
-    @objc private func didRecognizePanGesture(_ gestureRecognizer: UIPanGestureRecognizer) {
+    @objc private func handlePanGesture(_ pan: UIPanGestureRecognizer) {
         // The view shouldn't be draggable on iPads
         if showAsPopup { return }
 
-        let translation = gestureRecognizer.translation(in: contentView)
+        let translation = pan.translation(in: contentView)
         yPosition += translation.y
-        gestureRecognizer.setTranslation(CGPoint.zero, in: contentView)
-
-        if gestureRecognizer.state != .ended { return }
+        pan.setTranslation(CGPoint.zero, in: contentView)
 
-        let velocity = gestureRecognizer.velocity(in: contentView).y
-        let landingYPosition = yPosition + velocity / 10
+        if pan.state != .ended { return }
+        
+        let projectedVelocity = project(initialVelocity: pan.velocity(in: contentView).y,
+                                        decelerationRate: UIScrollView.DecelerationRate.normal.rawValue)
+        
         let nextYPosition: CGFloat
         
         let bottomHalfOfChildView = view.frame.height - (contentView.frame.height / 2)
+        let pannedPastHalfOfViewHeight = yPosition > bottomHalfOfChildView
+        
+        let closeByVelocity = projectedVelocity > contentView.frame.height
```

Comment:
The value of `projectedVelocity` is distance travelled, not specific endpoint. You'll have to add its current `yPosition` to this value to compare it against where you want the end point to be (in this case the contentView's maxY)